src/rgw/driver/sfs/user.cc

```diff
@@ -174,6 +174,15 @@ int SFSUser::create_bucket(
   placement_rule.storage_class = "STANDARD";
   info.placement_rule = placement_rule;
 
+  /*
+    Automatically enable versioning for the bucket when
+    the user creates a bucket with Object Lock enabled.
+    This adheres to AmazonS3 behavior. see:
+    https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lock-overview.html#object-lock-bucket-config
+  */
+  if (obj_lock_enabled)
+    info.flags = BUCKET_VERSIONED | BUCKET_OBJ_LOCK_ENABLED;
```

Comment:
shouldn't this be OR'ed with the existing flags? How do we know we're not squashing existing flags?

Reply:
This is a reasonable observation in general and if you think, we can enforce the safety ORing the existing flag.
I have doubts that in this case this could actually happen because:

1) this is a create operation, we are defining the flags inside this function.
2) I've copied this from `rados` implementation.